src/TestExplorer/TestRunner.ts

```diff
@@ -785,11 +787,13 @@ export class TestRunner {
                                             });
                                         subscriptions.push(terminateSession);
                                     } else {
+                                        await this.swiftTestOutputParser.close();
```

Comment:
You could add a dispose function that called `this.swiftTestOutputParser.close();` and add it to the subscription list instead of adding the call everywhere. Might be slightly cleaner